decidim-core/lib/decidim/core/test/shared_examples/comments_examples.rb

```diff
@@ -1039,9 +1039,46 @@
 
       it "shows a message indicating that comments are disabled" do
         visit resource_path
-        expect(page).to have_content("Comments are disabled at this time")
+        expect(page).to have_content("Comments are currently disabled, only administrators can reply or post new ones.")
         expect(page).to have_no_content("You need to be verified to comment at this moment")
       end
     end
   end
+
+  context "when authenticated administrator" do
+    let!(:organization) { create(:organization) }
+    let!(:user) { create(:user, :admin, :confirmed, organization:) }
+    let!(:comments) { create_list(:comment, 3, commentable:) }
+
+    before do
+      login_as user, scope: :user
+      visit resource_path
+    end
+
+    context "when comments are blocked" do
+      let(:active_step_id) { component.participatory_space.active_step.id }
+
+      before do
+        component.update!(step_settings: { active_step_id => { comments_blocked: true } })
+      end
+
+      it "allows admin to answer closed comments" do
+        visit resource_path
+        expect(page).to have_link("Comment")
+        page.find("a", text: "Comment").click
+        fill_in "Comment", with: "Test admin commenting in a closed comment."
+        click_on "Publish comment"
+        expect(page).to have_content("Test admin commenting in a closed comment.")
+
+        expect(page).to have_button("Reply")
+        first("button", text: "Reply").click
+        expect(page).to have_css(".comment-thread")
+        within first(".comment-thread") do
+          fill_in "Comment", with: "Test admin replying a closed comment."
+          click_on "Publish reply"
+        end
+        expect(page).to have_content("Test admin replying a closed comment.")
+      end
```

Comment:
Better move this concern inside the already created context for a blocked comment situation.

This way, you don't have to repeat the "before" block and the definition of the step.

Just add after the last example something like:

```ruby
context "and the user is an administrator" do
  it_behaves_like "can answer comments"
end

context "and the user has a role" do
  it_behaves_like "can ansser comments"
end
```

This with all the cases contemplated in the concern

Reply:
Done